src/core/reference/src/op/interpolate.cpp

```diff
@@ -63,29 +64,55 @@ InterpolateEvalHelper::InfoForGenericLinearONNXMode InterpolateEvalHelper::get_i
     }
 
     int64_t batch_size = input_shape[0];
-    int64_t num_channels = input_shape[1];
+    int64_t num_channels;
+    if (channel_last)
+        num_channels = input_shape[input_shape.size() - 1];
+    else
+        num_channels = input_shape[0];
```

Comment:
It was initialized with `1` before:
```suggestion
        num_channels = input_shape[1];
```